src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -188,19 +179,40 @@ public IPAddress(ReadOnlySpan<byte> address)
             }
             else if (address.Length == IPAddressParserStatics.IPv6AddressBytes)
             {
-                _numbers = new ushort[NumberOfLabels];
-
-                for (int i = 0; i < NumberOfLabels; i++)
-                {
-                    _numbers[i] = (ushort)(address[i * 2] * 256 + address[i * 2 + 1]);
-                }
+                _numbers = ReadUInt16NumbersFromBytes(address);
             }
             else
             {
                 throw new ArgumentException(SR.dns_bad_ip_address, nameof(address));
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ushort[] ReadUInt16NumbersFromBytes(ReadOnlySpan<byte> address)
+        {
+            if (!BitConverter.IsLittleEndian)
+            {
+                return MemoryMarshal.Cast<byte, ushort>(address).ToArray();
```

Comment:
This is non-portable cast. It assumes that that misaligned access to `ushort` is ok.

Would it be better to just depend on the non-vectorized path for big endian?

Reply:
Yup